internal/app/synchronizer/evaluator_client.go

```diff
@@ -145,6 +147,10 @@ func (ec *grcpEvaluatorClient) evaluate(ctx context.Context, pc <-chan []*pb.Mat
 			if err != nil {
 				return fmt.Errorf("failed to get response from evaluator client, desc: %w", err)
 			}
+			if !matchIDs[resp.GetMatchId()] {
```

Comment:
We can tell whether it was returned twice, or is an invalid id, the error messages should be distinct.